src/main/java/com/foodybuddy/service/impl/ApartmentServiceImpl.java

```diff
@@ -0,0 +1,290 @@
+		}
+		this.sessionFactory = sessionFactory;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see com.foodybuddy.service.ApartmentService#getById(java.lang.Integer)
+	 */
+	public Apartment getById(Integer id) throws Exception {
+		
+		Session session = null;
+		try {
+			if (id == null || id <= 0) {
+				if (id == null) {
+					throw new RuntimeException("ID is null");
+				} else {
+					throw new RuntimeException("ID is invalid");
+				}
+			}
+			
+			session = this.sessionFactory.openSession();
+			ApartmentDAO apartmentDAO = new ApartmentDAOImpl(session);
+			return apartmentDAO.getById(id);
+			
+		} catch (Exception exception) {
+			throw exception;
+		} finally {
+			if (session != null) {
+				session.close();
+			}
+		}
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see com.foodybuddy.service.ApartmentService#getAll()
+	 */
+	public List<Apartment> getAll() throws Exception {
+		Session session = null;
+		try {
+			session = this.sessionFactory.openSession();
+			ApartmentDAO apartmentDAO = new ApartmentDAOImpl(session);
+			List<Apartment> apartmentList = apartmentDAO.getAll();
+			if (apartmentList.size() == 0)
+				throw new RuntimeException("Empty Country List");
+			return apartmentList;
+		} catch (Exception exception) {
+			throw exception;
+		} finally {
+			if (session != null) {
+				session.close();
+			}
+		}
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see com.foodybuddy.service.ApartmentService#insert(java.lang.String,
+	 * java.lang.Integer)
+	 */
+	public Apartment insert(String name, Integer localityId, Integer blockNumber) throws TransactionException {
+
+		Session session = null;
+		Transaction transaction = null;
+
+		try {
+			
+			if (name == null || localityId == null || localityId <= 0 || name.trim().length() == 0 ) {
+
+				if (name == null) {
+					throw new RuntimeException("Name is null");
+				}
+
+				else if (localityId == null) {
+					throw new RuntimeException("LocalityId is null");
+				}
+
+				else if (localityId <= 0) {
+					throw new RuntimeException("LocalityId is invalid");
+				}
+
+				else {
+					throw new RuntimeException("Name is invalid");
+				}
+			}
+			session = this.sessionFactory.openSession();
+			transaction = session.beginTransaction();
+			transaction.setTimeout(10);
+
+			LocalityDAO localityDao = new LocalityDAOImpl(session);
+			Locality locality = localityDao.getById(localityId);
+
+			if (locality == null) {
+				throw ObjectNullException;
+			}
+
+			Apartment apartment = new Apartment();
+			apartment.setName(name);
+			apartment.setLocality(locality);
+			apartment.setBlockNumber(blockNumber);
+
+			ApartmentDAO apartmentDAO = new ApartmentDAOImpl(session);
+			apartmentDAO.insert(apartment);
+			transaction.commit();
+			return apartment;
+		}
+
+		catch (Exception exception) {
+			try {
+				if (transaction != null)
+					transaction.rollback();
+				throw TransactionFailureException;
+			}
+
+			catch (RuntimeException runtimeException) {
+				throw RollbackFailureException;
```

Comment:
Here the  Transaction rollback failed exception is thrown even if The transaction rollback is successfull